src/node_util.cc

```diff
@@ -38,6 +38,11 @@ using v8::Value;
 #undef V
 
 
+static void IsPrimitive(const FunctionCallbackInfo<Value>& args) {
+  args.GetReturnValue().Set(!args[0]->IsObject() && !args[0]->IsExternal());
```

Comment:
@benjamingr The whole point of this is to do it at the C++ layer instead of js because there is no reliable way to determine it in js (especially in a way that agrees with how v8 internally determines whether a value is a primitive or not).